### PR TITLE
Add PublishedHost tests in C#

### PR DESCRIPTION
**Pull request by bernardnormier**
Once approved and merged, I'll write equivalent tests for C++ and Java.

### CODE REVIEW

File: csharp/test/Ice/adapterDeactivation/AllTests.cs [hunk 1]
```diff
@@ -1,5 +1,7 @@
 // Copyright (c) ZeroC, Inc.
 
+using Ice.Internal;
+
 namespace Ice
 {
     namespace adapterDeactivation
```
Comment (bernardnormier) [inline csharp/test/Ice/adapterDeactivation/AllTests.cs:7]: Despite the name, these are really object adapter tests - not deactivation-specific.

**Review by pepone — APPROVED** [review]
Looks good, we can also add a test case for two endpoints + published host set:

`tcp -h *:ssl -h *` and you would get two endpoints with the published host.

File: csharp/test/Ice/adapterDeactivation/AllTests.cs [hunk 2]
```diff
@@ -94,6 +94,140 @@ public class AllTests : global::Test.AllTests
...
+                    {
+                        ObjectAdapter adapter = communicator.createObjectAdapter("PHAdapter");
+                        var publishedEndpoints = adapter.getPublishedEndpoints();
+                        test(publishedEndpoints.Length == 1);
+                        test(getUnderlying(publishedEndpoints[0].getInfo()) is IPEndpointInfo ipEndpointInfo &&
+                            ipEndpointInfo.host == "test.zeroc.com");
+                        adapter.destroy();
+                    }
+
+                    communicator.getProperties().setProperty("PHAdapter.Endpoints", "default -h 127.0.0.1");
```
Comment (externl) [inline csharp/test/Ice/adapterDeactivation/AllTests.cs:122]:
```suggestion
                    // Listening on loopback
                    communicator.getProperties().setProperty("PHAdapter.Endpoints", "default -h 127.0.0.1");
```

**Review by externl — APPROVED** [review]
Looks good, see comments.

File: csharp/test/Ice/adapterDeactivation/AllTests.cs [hunk 3]
```diff
@@ -186,6 +186,37 @@ public class AllTests : global::Test.AllTests
...
+                        test(getUnderlying(publishedEndpoints[1].getInfo()) is IPEndpointInfo ipEndpointInfo1 &&
+                            ipEndpointInfo1.host.Length > 0 &&
+                            ipEndpointInfo1.port != 12345);
+                        adapter.destroy();
+                    }
+
+                    communicator.getProperties().setProperty("PHAdapter.PublishedHost", "test.zeroc.com");
+                    {
+                        ObjectAdapter adapter = communicator.createObjectAdapter("PHAdapter"); // unset
+                        var publishedEndpoints = adapter.getPublishedEndpoints();
```
Comment (pepone) [inline csharp/test/Ice/adapterDeactivation/AllTests.cs:209]: Why is it legal to get the published endpoints, before activating the adapter? Here the ephemeral port would not be assigned yet, or am I missing something.

Comment (bernardnormier) [reply csharp/test/Ice/adapterDeactivation/AllTests.cs:209]: The ephemeral ports are assigned when you _create_ the object adapter. 

It's necessary for this common pattern:
```csharp
var adapter = communicator.createObjectAdapterWithEndpoints("MyAdapter", "tcp -h *"); // no port
var proxy = adapter.add(new GreeterService(), greeterIdentity); // port needs to be assigned at this point
adapter.activate();
```

**Pull request merged**